io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/writer/XLSInstanceWriter.java

```diff
@@ -112,6 +112,9 @@ else if (getContentType().getId().equals("eu.esdihumboldt.hale.io.xls.xlsx")) {
 
 		headerRowStrings = new ArrayList<String>();
 
+		boolean useSchema = getParameter(InstanceTableIOConstants.USE_SCHEMA).as(Boolean.class,
```

Comment:
Same as in the CSV writer - should use `getOptionalParameter(...)`

Reply:
The method `getOptionalParameter` is undefined for the type `CSVInstanceWriter` or `XLSInstanceWriter`. I think that for using this 'new' method I should then also add it into `AbstractIOProvider`, maybe like this:

         /**
	 * @return the parameter or a default value
	 * 
	 * @see IOProvider#getParameter(String)
	 */
	@Override
	public Value getOptionalParameter(String name) {
		Value value = parameters.get(name);
		if (value == null)
			value = Value.of(true);
		return value;
	}

isn't it? Am I making it more complex than what it really is?

Reply:
I believe, what Simon is referring to is as done in the below example, you don't need to override the method, just fetch the parameter value and it is null then return the default value. https://github.com/halestudio/hale/blob/44e54a9dfd067bdf51db427a50de9803213ffb1c/common/plugins/eu.esdihumboldt.hale.common.align/src/eu/esdihumboldt/hale/common/align/transformation/function/impl/AbstractTransformationFunction.java#L129

Reply:
I'm sorry, I mixed that up with the transformation functions, fine as it is.